code/obj/item/device/radios/headsets.dm

```diff
@@ -13,6 +13,36 @@
 	desc = "A standard-issue device that can be worn on a crewmember's ear to allow hands-free communication with the rest of the crew."
 	flags = FPRINT | TABLEPASS | CONDUCT | ONBELT
 	bicon_override = 1
+	var/upgraded
+
+	attackby(obj/item/R as obj, mob/user as mob)
+		if (istype(R, /obj/item/device/radio_upgrade))
+			if (upgraded)
+				boutput(user, "<span style=\"color:red\">This [src] already has a Wiretap Upgrade installed! What good could possibly come from having two?! </span>")
+				return
+			src.secure_frequencies = list(
+				"h" = R_FREQ_COMMAND,
+				"g" = R_FREQ_SECURITY,
+				"e" = R_FREQ_ENGINEERING,
+				"r" = R_FREQ_RESEARCH,
+				"m" = R_FREQ_MEDICAL,
+				"c" = R_FREQ_CIVILIAN,
+				)
+			src.secure_colors = list(
+				"h" = RADIOC_COMMAND,
+				"g" = RADIOC_SECURITY,
+				"e" = RADIOC_ENGINEERING,
+				"r" = RADIOC_RESEARCH,
+				"m" = RADIOC_MEDICAL,
+				"c" = RADIOC_CIVILIAN,
+				)
+			boutput(user, "<span style=\"color:blue\">Wiretap Radio Upgrade successfully installed in the [src].</span>")
+			playsound(src.loc ,"sound/items/Deconstruct.ogg", 80, 0)
+			set_secure_frequencies(src)
+			src.upgraded = 1
```

Comment:
I would set this at the beginning to avoid triggering this branch multiple times due to lag.

Reply:
Moved the src.upgraded = 1 (now called haswiretap) to the top, which is what I *think* you meant!